include/net/xia_ether.h

```diff
@@ -0,0 +1,36 @@
+#ifndef _NET_XIA_ETHER_H
+#define _NET_XIA_ETHER_H
```

Comment:
Pursuing readability is a laudable goal, but you are not using the header file properly.

First, the existential reason for header files is to export code to other modules. Yes, exporting code helps with organizing the code and, in turn, organized code improves readability. But the readability is gained from being able to break a C file in a number of C files, not in breaking a C file into a header and a C file.

Second, the header file `include/linux/netdevice.h` is being misused. For example, consider the following functions in this header file:
```
static void del_interface_addr(struct interface_addr *to_del)
int xia_ether_header_cache(const struct fib_xid_ether_main *mfxid, struct hh_cache *hh, __be16 type)
static void mfxid_hh_init(struct fib_xid_ether_main *mfxid)
```
Since they are not `inline` (and they shouldn't be since they are too big to be `inline`), they are being added to each C file that adds this header file. Right now, `net/xia/ppal_ether/main.c` is the only C file adding it, so it's okay. But, at the moment a second C file adds this header, these functions will be duplicated in both final object files. Not only is this going to grow the memory footprint of the kernel, but it can also introduce some hard to catch bugs.

Third, once all my feedback is incorporated into your code, there will be less code. I expect that `net/xia/ppal_ether/main.c` will become of a manageable size.